File: Cinema_groupe4_back/cinema_backend/src/main/java/com/intiformation/modele/Film.java
package com.intiformation.modele;

import java.sql.Date;
import java.sql.Time;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;

@Entity
public class Film {
	
	@Id
	private int visaExploitation;
	private String titre ; 
	private Time duree;
	private String realisateur; 
	private String producteur; 
	private String[] acteurs;
	private Date dateDeSortie;
	@ManyToMany(cascade=CascadeType.ALL)
	@JoinTable(name="Caracteristiques_join_Film", joinColumns = @JoinColumn(name = "IdFilm"), inverseJoinColumns=@JoinColumn(name="IdCarac"))
	private List<Caracteristiques> listeCaracteristiques;
	
	public Film() {
		super();
	}

	public Film(int visaExploitation, String titre, Time duree, String realisateur, String producteur, String[] acteurs,
			Date dateDeSortie, List<Caracteristiques> listeCaracteristiques) {
		super();
		this.visaExploitation = visaExploitation;
		this.titre = titre;
		this.duree = duree;
		this.realisateur = realisateur;
		this.producteur = producteur;
		this.acteurs = acteurs;
		this.dateDeSortie = dateDeSortie;
		this.listeCaracteristiques = listeCaracteristiques;
	}

	public int getVisaExploitation() {
		return visaExploitation;
	}

	public void setVisaExploitation(int visaExploitation) {
		this.visaExploitation = visaExploitation;
	}

	public String getTitre() {
		return titre;
	}

	public void setTitre(String titre) {
		this.titre = titre;
	}

	public Time getDuree() {
		return duree;
	}

	public void setDuree(Time duree) {
		this.duree = duree;
	}

	public String getRealisateur() {
		return realisateur;
	}

	public void setRealisateur(String realisateur) {
		this.realisateur = realisateur;
	}

	public String getProducteur() {
		return producteur;
	}

	public void setProducteur(String producteur) {
		this.producteur = producteur;
	}

	public String[] getActeurs() {
		return acteurs;
	}

	public void setActeurs(String[] acteurs) {
		this.acteurs = acteurs;
	}

	public Date getDateDeSortie() {
		return dateDeSortie;
	}

	public void setDateDeSortie(Date dateDeSortie) {
		this.dateDeSortie = dateDeSortie;
	}

	public List<Caracteristiques> getListeCaracteristiques() {
		return listeCaracteristiques;
	}

	public void setListeCaracteristiques(List<Caracteristiques> listeCaracteristiques) {
		this.listeCaracteristiques = listeCaracteristiques;
	}
	
	
	
	
	
	
	
	
	
	
	
	
	
}

File: Cinema_groupe4_front/cinema_front/src/app/model/film.ts
export class Film{
    private _visaExploitation : number;
    private _titre : string;
    private _duree : Date;
    private _realisateur : string;
    private _producteur : string;
    private _acteur : string[];
    private _carcteristique : string[];

    get visaExploitation() : number {
        return this._visaExploitation;
    }

    set visaExploitation(value : number) {
        this._visaExploitation = value;
    }

    get titre() : string {
        return this._titre;
    }

    set titre(value : string) {
        this._titre = value;
    }

    get duree() : Date {
        return this._duree;
    }

    set duree(value : Date) {
        this._duree = value;
    }

    get realisateur() : string {
        return this._realisateur;
    }

    set realisateur(value : string) {
        this._realisateur = value;
    }

    get producteur() : string {
        return this._producteur;
    }

    set producteur(value : string) {
        this._producteur = value;
    }

    get acteur() : string[] {
        return this._acteur;
    }

    set acteur(value : string[]) {
        this._acteur = value;
    }

    get carcteristique() : string[] {
        return this._carcteristique;
    }

    set carcteristique(value : string[]) {
        this._carcteristique = value;
    }

    constructor(visaExploitation : number, titre : string, duree : Date, realisateur : string, producteur : string, acteur : string[], carcteristique : string[]){
        this._visaExploitation = visaExploitation;
        this._titre = titre;
        this._duree = duree;
        this._realisateur = realisateur;
        this._producteur = producteur;
        this._acteur = acteur;
        this._carcteristique = carcteristique;
    }
}